source/beamline/AMSlit.h

```diff
@@ -0,0 +1,144 @@
+#ifndef AMSLIT_H
+#define AMSLIT_H
+
+#include "beamline/AMControl.h"
+#include "beamline/AMSlitControl.h"
+#include "beamline/AMControlSet.h"
+
+#define AMSLIT_CANNOT_OPEN 39810
+#define AMSLIT_CANNOT_CLOSE 39811
+
+class AMSlit : public AMControl
```

Comment:
It's true that the two classes do share firstBlade_, secondBlade_, and orientation map class variables, and maybe it's worth modifying AMSlit to inherit from AMSlitControl to take advantage of this. Beyond having those variables in common, though, the purpose of each class diverges: AMSlitControl is meant to provide the equations for calculating gap and center for different blade orientations, while AMSlit is the main object that provides access to the gap, center, and blade controls. AMSlitControls are measurable, AMSlit isn't. To me, it makes a little more sense to keep them separate but an argument could be made either way.